src/LdswScraper/HostScraper.cs

```diff
@@ -0,0 +1,230 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using VDS.RDF;
+
+namespace LdswScraper;
+
+public class HostScraper
+{
+    private readonly HttpClient _httpClient;
+    private readonly ScraperConfig _config;
+    private readonly string _hostname;
+    private readonly List<ScrapeTask> _tasks;
+
+    public HostScraper(HttpClient httpClient, ScraperConfig config, string hostname, List<ScrapeTask> tasks)
+    {
+        _httpClient = httpClient;
+        _config = config;
+        _hostname = hostname;
+        _tasks = tasks;
+    }
+
+    public async Task RunAsync()
+    {
+        foreach (var task in _tasks)
+        {
+            await ProcessTaskAsync(task);
+            await Task.Delay(TimeSpan.FromSeconds(_config.DelaySeconds));
+        }
+    }
+
+    private async Task ProcessTaskAsync(ScrapeTask task)
+    {
+        Console.WriteLine($"Processing {task.Uri} ({task.Type})");
+
+        switch (task.Type)
+        {
+            case TaskType.Conneg:
+                await ProcessConnegAsync(task);
+                break;
+            case TaskType.Exact:
+                await ProcessExactAsync(task);
+                break;
+            case TaskType.Shex:
+                await ProcessShexAsync(task);
+                break;
+        }
+    }
+
+    private async Task ProcessConnegAsync(ScrapeTask task)
```

Comment:
@jules make one Console.WriteLine call for the whole conneg task to avoid output buffering messing up the output when running in parallel.